plugins/modules/user_role_workflow_manager.py

```diff
@@ -1661,23 +1666,29 @@ def create_role(self, role_params):
             - Logs the provided role parameters and the received API response.
             - Returns the API response from the "create_role" function.
         """
-        try:
-            self.log("Create role with role_info_params: {0}".format(str(role_params)), "DEBUG")
-            response = self.dnac._exec(
-                family="user_and_roles",
-                function="add_role_api",
-                op_modifies=True,
-                params=role_params,
-            )
-            self.log("Received API response from create_role: {0}".format(str(response)), "DEBUG")
-            self.created_role.append(role_params.get("role"))
-            return response
 
-        except Exception as e:
-            self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
-            error_message = "The Catalyst Center user '{0}' does not have the necessary permissions to 'create a role' through the API.".format(
-                self.payload.get("dnac_username"))
-            return {"error_message": error_message}
+        if self.compare_dnac_versions(self.get_ccc_version(), "2.3.7.6") >= 0:
+            try:
+                self.log("Create role with role_info_params: {0}".format(str(role_params)), "DEBUG")
+                response = self.dnac._exec(
+                    family="user_and_roles",
+                    function="add_role_api",
+                    op_modifies=True,
+                    params=role_params,
+                )
+                self.log("Received API response from create_role: {0}".format(str(response)), "DEBUG")
+                self.created_role.append(role_params.get("role"))
+                return response
+
+            except Exception as e:
+                self.log("Unexpected error occurred: {0}".format(str(e)), "ERROR")
+                error_message = "The Catalyst Center user '{0}' does not have the necessary permissions to 'create a role' through the API.".format(
+                    self.payload.get("dnac_username"))
+                return {"error_message": error_message}
+
+        error_message = "The specified version '{0}' does not have the 'add_role_api' functionality. Supported version(s) '2.3.7.6'.".format(
```

Comment:
"The specified version '{0}' does not have the 'add_role_api' functionality. "
    "Supported version(s) from '2.3.7.6' onwards."

Reply:
Updated the code accordingly.